Ensure tests can be passed with the 'Very strong' password policy

abashurov opened the pull request:
Currently, tests only pass with weaker policies.

Additional changes in this PR are to allow passing '&' symbol in the password entity and to enable tests on licenses with unlimited customers/domains/resellers.

mrsombre commented on tests/DatabaseTest.php:
```diff
@@ -52,12 +54,12 @@ public function testUpdateUser()
             'id' => $user->id,
             'login' => 'test_user2',
-            'password' => 'setup2Q',
+            'password' => PasswordProvider::getPassword() . '1',
```
I think we can set default password for this user. No idea why we need change it for this user.

abashurov replied:
Updated accordingly to the suggestion

mrsombre replied:
Okay, i'll check it tomorrow. Other things looks good now.

mrsombre commented on tests/SiteTest.php:
```diff
@@ -19,7 +19,7 @@ protected function setUp()
         $keyInfo = static::$_client->server()->getKeyInfo();
 
-        if ((int)$keyInfo['lim_dom'] < 2) {
+        if ((int)$keyInfo['lim_dom'] > -1 && (int)$keyInfo['lim_dom'] < 2) {
```
Please, create method/trait for this check (code duplication)

abashurov replied:
Moved out into the Utility to act as a method of the separate class

mrsombre commented on tests/TestCase.php:
```diff
@@ -60,7 +62,7 @@ protected static function _createWebspace()
             ], [
                 'ftp_login' => "u{$id}",
-                'ftp_password' => $password,
+                'ftp_password' => PasswordProvider::getPassword() . $password,
```
Just PasswordProvider::getPassword()

abashurov replied:
Updated accordingly to the suggestion

mrsombre commented on src/Api/Client.php:
```diff
@@ -318,7 +318,7 @@ protected function _arrayToXml(array $array, SimpleXMLElement $xml, $parentEl =
                 $this->_arrayToXml($value, $this->_isAssocArray($value) ? $xml->addChild($el) : $xml, $el);
             } else {
-                $xml->addChild($el, $value);
+                $xml->$el = $value;
```
I suppose there must be escape function like this
$value = htmlspecialchars($value, ENT_QUOTES | ENT_XML1);

abashurov replied:
There is an escape function, but API endpoint does not unescape this value: it gets stored in the database as-is and leaves account with the 'qwerty&amp;' password instead of the 'qwerty&'

sibprogrammer commented:
In general I highly against of mixing ~3 different things into single commit. Moreover escaping topic is more tricking than it seems at a first glance.

abashurov commented:
> In general I highly against of mixing ~3 different things into single commit. Moreover escaping topic is more tricking than it seems at a first glance.

Removed '&' from passwords until escaping entities is sorted out and split the PR into different commits
